.github/workflows/release.yaml

```diff
@@ -17,45 +17,86 @@ name: release
 on:
   push:
     tags:
-      - 'v*'
+      - 'v[0-9]+\.[0-9]+\.[0-9]+(-build\.[0-9]+)?'
 
 jobs:
-  release:
+  check-if-main:
```

Comment:
I feel that the `check-if-main` probably isn't necessary now that we're running tests on release again.